app/ide-desktop/lib/dashboard/src/authentication/src/authentication/config.ts

```diff
@@ -0,0 +1,128 @@
+/** @file Configuration for the authentication module of the Dashboard. */
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/** AWS region in which our Cognito pool is located. */
+export const AWS_REGION: AwsRegion = 'eu-west-1'
+/** URL used as the OAuth redirect when running in the desktop app. */
+export const DESKTOP_REDIRECT: OAuthRedirect = 'enso://auth'
+/** Complete list of OAuth scopes used by the app. */
+export const OAUTH_SCOPES: OAuthScope[] = ['email', 'openid']
+/** OAuth response type used in the OAuth flows. */
+export const OAUTH_RESPONSE_TYPE: OAuthResponseType = 'code'
+
+
+
+// =============
+// === Types ===
+// =============
+
+/** The AWS region in which our Cognito pool is located. This is always set to `eu-west-1` because
+ * that is the only region in which our Cognito pools are currently available in. */
+type AwsRegion = 'eu-west-1'
+/** ID of the "Cognito user pool" that contains authentication & identity data of our users.
+ *
+ * This is created automatically by our Terraform scripts when the backend infrastructure is
+ * created. Look in the `enso-org/cloud-v2` repo for details. */
+type UserPoolId = string
```

Comment:
not sure if this has been asked before, feel free to close it if it has.

`UserPoolId` is exactly the same as `string`, so this doesn't offer any additional type safety - doing this is mainly useful for documentation (= it makes it clearer what `id` is in `{ id: UserPoolId }`)

i'm using brands (poor man's newtypes: `string & { _brand: "UserPoolId"; }`) in one of my PRs, which works great when:
- they come from an API, and
- you shouldn't be operating on them (they turn back to regular `string`s when you do anything to them), and
- you pass them around a lot so the extra type safety is useful

Reply:
I'm new to TypeScript so I wasn't aware that this was an option when providing type safety. I like it. I'll convert this code to use brands.